cas-server-support-osf/src/main/java/io/cos/cas/web/flow/OpenScienceFrameworkLoginHandler.java

```diff
@@ -171,6 +189,17 @@ private boolean isInstitutionLogin(final RequestContext context) {
         return "institution".equals(campaign);
     }
 
+    /**
+     * Check if the request is ORCiD login redirect.
+     *
+     * @param context the request context
+     * @return true if `redirectOrcid=true` is present in the request parameters
+     */
+    private boolean isOrcidRedirect(final RequestContext context) {
+        final String orcidRedirect = context.getRequestParameters().get("redirectOrcid");
+        return orcidRedirect != null && "true".equals(orcidRedirect.toLowerCase());
```

Comment:
Consider using `campaign != null && "institution".equals(campaign.toLowerCase())` for `isInstitutionLogin()`.